### DISCUSSION TITLE
Customizing the face for a specific semantic token

### DISCUSSION THREAD

**Discussion by itamarst** [emacs-lsp/lsp-mode#3664] [Q&A]
Hi,

I'm trying to reproduce https://veykril.github.io/posts/semantic-unsafe/ in lsp-mode. TL;DR is in VSCode they set:

```json
"editor.semanticTokenColorCustomizations": {
    "rules": {
        "*.unsafe:rust": "#eb5046"
    }
}
```

In `lsp-mode`, there's documentation for enabling semantic tokens, but none for customizing the faces. I've tried:

```elisp
(setq lsp-enable-semantic-highlighting 't)
(setq lsp-semantic-tokens-enable 't)
(add-to-list 'lsp-semantic-token-faces '("unsafe" . rust-unsafe-face))
```

But that doesn't seem to work.

**Comment by ericdallo** [comment]
c/c @sebastiansturm

**Reply by alanz** [reply]
See also https://github.com/emacs-lsp/lsp-mode/pull/3668, which needs a small update prior to landing

**Comment by flodiebold** [comment]
I think you need to set `lsp-semantic-token-modifier-faces` instead of `lsp-semantic-token-faces`. Haven't tested it recently though.
